client/state/site-settings/actions.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	SITE_SETTINGS_RECEIVE,
+	SITE_SETTINGS_REQUEST,
+	SITE_SETTINGS_REQUEST_FAILURE,
+	SITE_SETTINGS_REQUEST_SUCCESS,
+	SITE_SETTINGS_SAVE,
+	SITE_SETTINGS_SAVE_FAILURE,
+	SITE_SETTINGS_SAVE_SUCCESS,
+	SITE_SETTINGS_UPDATE
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that site settings have been received.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The site settings object
+ * @return {Object}        Action object
+ */
+export function receiveSiteSettings( siteId, settings ) {
+	return {
+		type: SITE_SETTINGS_RECEIVE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that some site settings have been update.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The updated site settings
+ * @return {Object}        Action object
+ */
+export function updateSiteSettings( siteId, settings ) {
+	return {
+		type: SITE_SETTINGS_UPDATE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action thunk which, when invoked, triggers a network request to
+ * retrieve site settings
+ *
+ * @param  {Number} siteId Site ID
+ * @return {Function}      Action thunk
+ */
+export function requestSiteSettings( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SITE_SETTINGS_REQUEST,
+			siteId
+		} );
+
+		return new Promise( ( resolve, reject ) => {
```

Comment:
You're better off updating the undocumented method to return the result of `this.wpcom.req`, which itself is a promise, avoiding the need to create your own here:

https://github.com/Automattic/wp-calypso/blob/d980735/client/lib/wpcom-undocumented/lib/undocumented.js#L291

i.e. if you update that line to:

```js
return this.wpcom.req[ method ]( {
```

You can change here to:

```js
wpcom.undocumented().settings( siteId ).then( ( data ) => {
```